ebean-core/src/test/java/org/tests/query/TestQueryIsNull.java

```diff
@@ -17,90 +18,90 @@
   public void queryShouldContainIsNullOnColumn() {
     ResetBasicData.reset();
 
-    Query<Order> query = Ebean.find(Order.class).where().isNull("customerName").query();
+    Query<Order> query = DB.find(Order.class).where().isNull("customerName").query();
     query.findList();
 
-    assertTrue(query.getGeneratedSql().contains("name is null"));
+    assertThat(query.getGeneratedSql()).contains("name is null");
   }
 
   @Test
   public void isNotNull_when_OneToMany_expect_existsSubquery() {
     ResetBasicData.reset();
 
-    Query<Order> query = Ebean.find(Order.class).where().isNotNull("details").query();
+    Query<Order> query = DB.find(Order.class).where().isNotNull("details").query();
     query.findList();
 
-    assertTrue(query.getGeneratedSql().contains(" where exists (select 1 from o_order_detail x where x.order_id = t0.id)"));
+    assertThat(query.getGeneratedSql()).contains(" where exists (select 1 from o_order_detail x where x.order_id = t0.id and x.id > 0)");
   }
 
   @Test
   public void isNotEmpty_when_OneToMany_expect_existsSubquery() {
     ResetBasicData.reset();
 
-    Query<Order> query = Ebean.find(Order.class).where().isNotEmpty("details").query();
+    Query<Order> query = DB.find(Order.class).where().isNotEmpty("details").query();
     query.findList();
 
-    assertTrue(query.getGeneratedSql().contains(" where exists (select 1 from o_order_detail x where x.order_id = t0.id)"));
+    assertThat(query.getGeneratedSql()).contains(" where exists (select 1 from o_order_detail x where x.order_id = t0.id and x.id > 0)");
   }
 
   @Test
   public void isNull_when_OneToMany_expect_notExistsSubquery() {
     ResetBasicData.reset();
 
-    Query<Order> query = Ebean.find(Order.class).where().isNull("details").query();
+    Query<Order> query = DB.find(Order.class).where().isNull("details").query();
     query.findList();
 
-    assertTrue(query.getGeneratedSql().contains(" where not exists (select 1 from o_order_detail x where x.order_id = t0.id)"));
+    assertThat(query.getGeneratedSql()).contains(" where not exists (select 1 from o_order_detail x where x.order_id = t0.id and x.id > 0)");
   }
 
   @Test
   public void isEmpty_when_OneToMany_expect_notExistsSubquery() {
     ResetBasicData.reset();
 
-    Query<Order> query = Ebean.find(Order.class).where().isEmpty("details").query();
+    Query<Order> query = DB.find(Order.class).where().isEmpty("details").query();
     query.findList();
 
-    assertTrue(query.getGeneratedSql().contains(" where not exists (select 1 from o_order_detail x where x.order_id = t0.id)"));
+    assertThat(query.getGeneratedSql()).contains(" where not exists (select 1 from o_order_detail x where x.order_id = t0.id and x.id > 0)");
```

Comment:
@rbygrave what do you think? We must add the extra where here and the old code was wrong IMHO. 
The fix may introduce breaking changes for other people

Reply:
Yes, it looks like the old code was wrong.  It looks we should bump to 12.10.x with this change.